### ISSUE TITLE
Question for mode of thread

### ISSUE THREAD

**Issue by Lym1028987502** [thieu1995/mealpy#72]
Thank you for providing the useful codes. There is a question when I use the the mode of thread to solve the problem. When I use the mode of thread to solve the problem and  count the number of active threading, I  set the max_workers=4.  When the GA or the some other algorithms are used, the number of threading is always printed as 5 except PSO/CA/EOA/BRO etc, the numuber of threading for which are first showed as 5 first and then 1 .  It means that only 1 threading is working later when the PSO/CA/EOA/BRO are used. I only test about 20 algorithms, and PSO/CA/EOA/BRO are found for the problem. Could you help me solve the problem? especially for  PSO.
    the example is shown below.
def fitness_func(solution):
    print(threading.active_count())
    return np.sum(solution ** 2)
model = PSO.BasePSO(problem_dict, epoch=100, pop_size=50)
best_position, best_fitness_value = model.solve(mode="thread")
    Besides, the max_worker=4 is used in optimizer.py:
    # with parallel.ProcessPoolExecutor(self.n_workers) as executor:
    with parallel.ProcessPoolExecutor(max_workers=4) as executor:
    Looking farward to your reply, thank you!

**Comment by Lym1028987502** [comment]
Thanks for your answer. I have installed the library via pip.  And I have tested my code in the method you provided, geting the same results. Additionally, I think that the PSO is not requried to calculated the fitness function alone, for the number of thread  shows 5 and the 5 threads are working at the beginning. It comfuses me that only 1 thread is working later. I hope that you can help me solve the problem, if you are interested in it. Thank you!

**Comment by thieu1995** [comment]
Hi @Lym1028987502 ,

I have looked back at the code. There are a few algorithms that actually cannot be run in parallel mode. Means that algorithm doesn't use the function update_target_wrapper_population(), for example: PSO, CA, EOA, BRO.

The first print() is 1 thread because every time you try an algorithm, it will try to calculate the fitness function to see whether your fitness return value or not (valid or not).

From the second print() is n+1 for every algorithm, even with the algorithm that cannot be run in parallel mode. Because the initialization phase is starting, the function create_population() is called. n is the number of threads you set, and +1 is the main thread.

From the next print():
	+ If the algorithm doesn't use update_target_wrapper_population(), then it will show 1 thread 
	+ If the algorithm can run in parallel mode, then it will show n+1

I did not implement the PSO in parallel mode due to comparing local history with the new position operator. Not every algorithm can be parallel without changing its original structure.

Hope it helps
